sound/soc/sof/ipc4-priv.h

```diff
@@ -13,18 +13,27 @@
 #include <sound/sof/ext_manifest4.h>
 #include "sof-priv.h"
 
+enum sof_ipc4_mtrace_type {
+	SOF_IPC4_MTRACE_NOT_AVAILABLE = 0,
+	SOF_IPC4_MTRACE_INTEL_TGL,
```

Comment:
the trace design is based on CAVS version. This PR is applicable to cavs2x+ platform, such as icelake and tigerlake, mtl. For SKL, APL, CML, we need cavs1x.  Can we rename it to SOF_IPC4_MTRACE_INTEL_CAVS2X ?

Reply:
@RanderWang, OK, let's change to `SOF_IPC4_MTRACE_INTEL_CAVS2X`.
As a sidenote: it is unclear how this maps for me: it looks like that APL and CNL have incompatible logging and it depends where you are looking, SKL is either uses APL or CNL type of mtrace.
CAVS1x will not cut it as we have at least two types on there.
But let's care about that if we ever implement support for those...

Reply:
yes,  we should use CAVS15 APL and CAVS18 CNL,

Reply:
Thanks, I think the only difference is the message itself to enable the logging. library vs core and APL does not have *_timer_period parameters while CNL have.
I'll still left them out until it is clear and tested.